Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```diff
@@ -24,66 +47,177 @@ public partial class GestureManager : Singleton<GestureManager>
         /// set the override focused object.
         /// If its null, then the gazed at object will be selected.
         /// </summary>
-        public GameObject OverrideFocusedObject
-        {
-            get; set;
-        }
-
+        public GameObject OverrideFocusedObject { get; set; }
+        
         /// <summary>
         /// Gets the currently focused object, or null if none.
         /// </summary>
-        public GameObject FocusedObject
+        public GameObject FocusedObject { get; private set; }
+
+        /// <summary>
+        /// Whether or not a manipulation gesture is currently in progress
+        /// </summary>
+        public bool ManipulationInProgress { get; private set; }
+
+        /// <summary>
+        /// The offset of the hand from its position at the beginning of 
+        /// the currently active manipulation gesture, in world space.  Not valid if
+        /// a manipulation gesture is not in progress
+        /// </summary>
+        public Vector3 ManipulationOffset { get; private set; }
+
+        /// <summary>
+        /// The world space position of the hand being used for the current manipulation gesture.  Not valid
+        /// if a manipulation gesture is not in progress.
+        /// </summary>
+        public Vector3 ManipulationHandPosition
         {
-            get { return focusedObject; }
+            get
+            {
+                Vector3 handPosition = Vector3.zero;
+                currentHandState.properties.location.TryGetPosition(out handPosition);
+                return handPosition;
+            }
         }
 
         private GestureRecognizer gestureRecognizer;
-        private GameObject focusedObject;
+        private GestureRecognizer manipulationRecognizer;
+
+        private bool HandPressed { get { return pressedHands.Count > 0; } }
+        private HashSet<uint> pressedHands = new HashSet<uint>();
+
+        private InteractionSourceState currentHandState;
 
         void Start()
         {
+            InteractionManager.SourcePressed += InteractionManager_SourcePressed;
```

Comment:
I struggled with this one, and I share your concerns about having InteractionManager event handlers scattered about.  Here's why I chose to put this in GestureManager:

In order for this manipulation algorithm to operate we need the actual world space position of the hand when the gesture begins.  Unfortunately, the Manipulation gesture doesn't provide any way to determine which hand is performing the gesture, making it difficult to access this information since it's possible that multiple hands could be pressed and multiple gestures could be occurring at once.  To work around this we subscribe to SourcePressed/Released/Updated/Lost directly in GestureManager and track the first pressed hand until that hand is no longer pressed.  We call this the "current" hand for any gestures that GestureManager exposes.  This limits GestureManager to only one gesture at a time, but it significantly reduces complexity for anyone who needs to access additional hand information which the gesture itself doesn't provide.

If we were to remove this code from GestureManager we would probably just end up with very similar code in GestureManipulator to solve the same problem, and anyone who wanted to use the manipulation gesture in a different way but still access data about the hand involved would have to implement this themselves.

This method of associating hands with gestures is definitely imperfect, but it's been well tested and has performed well (this is what we shipped in 3D Viewer), which is why I implemented it here.

If the only objection is "It would be nice to have all the SourceX events in the same place" I think the pros of the current implementation significantly outweigh the cons.  

@NeerajW What do you think?